src/main/java/com/gtnewhorizons/angelica/transform/HUDCachingTransformer.java

```diff
@@ -0,0 +1,75 @@
+package com.gtnewhorizons.angelica.transform;
+
+import com.gtnewhorizons.angelica.loading.AngelicaTweaker;
+import net.minecraft.launchwrapper.IClassTransformer;
+import org.objectweb.asm.ClassReader;
+import org.objectweb.asm.ClassWriter;
+import org.objectweb.asm.Opcodes;
+import org.objectweb.asm.tree.AnnotationNode;
+import org.objectweb.asm.tree.ClassNode;
+import org.objectweb.asm.tree.InsnList;
+import org.objectweb.asm.tree.InsnNode;
+import org.objectweb.asm.tree.JumpInsnNode;
+import org.objectweb.asm.tree.LabelNode;
+import org.objectweb.asm.tree.LdcInsnNode;
+import org.objectweb.asm.tree.MethodInsnNode;
+import org.objectweb.asm.tree.MethodNode;
+
+
+public class HUDCachingTransformer implements IClassTransformer {
+    private static final String HUDCaching = "com/gtnewhorizons/angelica/hudcaching/HUDCaching$Hooks";
+
+    @Override
+    public byte[] transform(String name, String transformedName, byte[] basicClass) {
+
+        final ClassReader cr = new ClassReader(basicClass);
+        final ClassNode cn = new ClassNode();
+        cr.accept(cn, 0);
+
+        final boolean changed = transformClassNode(transformedName, cn);
+        if (changed) {
+            ClassWriter cw = new ClassWriter(ClassWriter.COMPUTE_FRAMES);
```

Comment:
use the mixin class writer if you use compute frames `org.spongepowered.asm.transformers.MixinClassWriter`. The default class writer triggers class loading by using Class.forName when you use compute frames

Reply:
Does not work
![image](https://github.com/user-attachments/assets/2feb8e55-060d-435a-b37a-5d2974767b2c)

Reply:
you need switch to use the ASM objects included into the mixins library, look at this commit : https://github.com/GTNewHorizons/Battlegear2/pull/23